### DISCUSSION TITLE
How to forward a CMake cache variable when using `sysbuild`?

### DISCUSSION THREAD

**Discussion by TorstenRobitzki** [zephyrproject-rtos/zephyr#79869] [Q&A]
I try to use `sysbuild` to build a firmware for nrf5340. I need a CMake cache variable in the `CMakeLists.txt` in my application (to emulate multiple build targets).

I use `CMake` to create the build system and want to have `CACHE_VAR` to be set to `Value` with in `CMakeLists.txt`

    > cmake -DBOARD=<board> -DAPP_DIR=.. -DCACHE_VAR=Value ../../zephyr/share/sysbuild

Does not work (variable not set in `CMakeLists.txt`). According to the documentation, I thought, changing the variable name to either `app_CACHE_VAR` or `APP_CACHE_VAR` should do the trick, but unfortunately that's not the case.

How to forward a CMake cache variable when using `sysbuild`?

**Comment by merethan** [comment]
Hey Torsten,

There's an undocumented feature for this purpose, called zephyr_get(). The source tree is littered with it but as of yet no documentation has been written for it.

In your application `CMakeLists.txt`:

```
find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})

test_sysbuild()

# Must appear after find_package(Zephyr)
zephyr_get(CACHE_VAR SYSBUILD GLOBAL)
```

...and you're in business.

**Reply by TorstenRobitzki** [reply]
Hi @merethan 
will try that! Thank you!
